### PR TITLE
req: fix Coverity 1485137 Explicit null dereference

### PR DESCRIPTION
**Pull request by paulidale**
Add a check for a non-existent file name when specifying params via file.
Add a check for a failure to determine key type.

### CODE REVIEW

File: apps/req.c [hunk 1]
```diff
@@ -1569,6 +1575,10 @@ static EVP_PKEY_CTX *set_keygen_ctx(const char *gstr,
         }
         if (keytype == NULL) {
             keytype = EVP_PKEY_get0_type_name(param);
+            if (keytype == NULL) {
+                BIO_puts(bio_err, "Unable to determine key type\n");
```
Comment (slontis) [inline apps/req.c:1579]: Looks like a memory leak will happen here now? (param)

File: apps/req.c [hunk 2]
```diff
@@ -1522,6 +1522,12 @@ static EVP_PKEY_CTX *set_keygen_ctx(const char *gstr,
 
         if (strncmp(gstr, "param", len) == 0) {
             expect_paramfile = 1;
+            if (p == NULL) {
+                BIO_printf(bio_err,
+                           "Parameter file requested but no path given: %s\n",
+                           gstr);
```
Comment (slontis) [inline apps/req.c:1528]: What an unfortunate name (gstring).

File: apps/req.c (hunk 1 above)
Comment (paulidale) [reply apps/req.c:1579]: Good point.  I don't think that this path is actually possible but better to be safe and check for it.

**Comment by paulidale** [comment]
updated.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the reviews and comments.

**Pull request closed without merging**